Firestore/core/src/nanopb/reader.h

```diff
@@ -45,6 +45,9 @@ namespace nanopb {
  */
 class Reader {
  public:
+  Reader() = default;
+  explicit Reader(util::ReadContext context) : context_(std::move(context)) {
```

Comment:
I'm surprised this is necessary/works. 

The way you're calling this makes a copy of the ReadContext the caller supplies. This suggests that the local `ReadContext` they're keeping is irrelevant because it's disconnected from what's actually being passed down.

Are you sure this actually works?

If you're able to change the Serializer to just use the `ReadContext*` instead for the methods you need, then you can just remove these additions and not solve this mystery :-).